frontend/src/component/ui/Toast.spec.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Toast } from './Toast';
+
+describe('toast', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  test('should render without crash', () => {
+    render(<Toast toastMessage={'Toast message.'} />);
+  });
+
+  test('should show content correctly when triggered to show', () => {
+    const toastRef = React.createRef<Toast>();
+    const { container } = render(
+      <Toast ref={toastRef} toastMessage={'Toast message.'} />
+    );
+
+    expect(container.textContent).not.toContain('Toast message.');
+    toastRef.current!.show();
+    jest.runAllTimers();
+    expect(container.textContent).toContain('Toast message.');
+  });
+
+  test('should hide content correctly when triggered to hide', () => {
+    const toastRef = React.createRef<Toast>();
+    const { container } = render(
+      <Toast ref={toastRef} toastMessage={'Toast message.'} />
+    );
+
+    toastRef.current!.show();
+    jest.runAllTimers();
+    expect(container.textContent).toContain('Toast message.');
+
+    toastRef.current!.hide();
+    jest.runAllTimers();
+    expect(container.textContent).not.toContain('Toast message.');
+  });
+
+  test('should automatically hide content after delay', () => {
+    const toastRef = React.createRef<Toast>();
+    const { container } = render(
+      <Toast ref={toastRef} toastMessage={'Toast message.'} />
+    );
+
+    expect(container.textContent).not.toContain('Toast message.');
+    toastRef.current!.showAndHide(2000);
+
+    jest.advanceTimersByTime(1000);
+    expect(container.textContent).toContain('Toast message.');
+
+    jest.advanceTimersByTime(1000);
+    expect(container.textContent).not.toContain('Toast message.');
+
+    jest.clearAllTimers();
+  });
+
+  test('second showAndHide call should not affect first toast', () => {
```

Comment:
I think this one should actually effect the first toast when we have two notifications overlapping. What are your thoughts?

Reply:
Ohkay, do we need to extend the delay for closing the toast or show multiple toast messages?

Reply:
@rohithbalaji123 How about overriding the first Toast message and time out? I am not sure what's the best approach.

Reply:
@byliuyang Yeah, showing multiple toast messages doesn't sound right as the clipboard is overwritten by the new url.